integration-tests/src/main/scala/io/stryker4s/validator/OrderValidator.scala

```diff
@@ -0,0 +1,30 @@
+package io.stryker4s.validator
+
+import cats._
+import io.stryker4s.entity.Drink
+import io.stryker4s.service.OrderNotAllowed
+
+import scala.language.higherKinds
+
+object OrderValidator {
+  def apply[F[_]: Monad](): OrderValidator[F] = new OrderValidator[F]()
+}
+
+class OrderValidator[F[_]: Monad] {
+
+  def allowedToOrderDrinks(age: Int, drinks: List[Drink]): Either[OrderNotAllowed, Unit] = {
```

Comment:
Isn't this just a boolean with a name? 🤔

Reply:
In a sense yes, but you have to keep in mind that we are in a functional context where we want to for comp over the results. This makes that possible.

Also, keep in mind its custom that we collect all the errors first and return the whole reason that's rejected instead of 1 by 1 (fail fast) in some cases that make sense. In this case, it doesn't make sense to fail fast in my opinion (which is fail fast at this moment), but this implementation is close to a cats "validated" approach were we validate everything first then return the errors. but since i only have one validation rule as of now i didn't come around to build that yet.